libqtile/core/manager.py

```diff
@@ -405,25 +405,33 @@ def reconfigure_screens(self, ev: Any = None) -> None:
     def paint_screen(self, screen: Screen, image_path: str, mode: str | None = None) -> None:
         self.core.painter.paint(screen, image_path, mode)
 
-    def process_key_event(self, keysym: int, mask: int) -> None:
+    def process_key_event(self, keysym: int, mask: int) -> tuple[Key | KeyChord | None, bool]:
         key = self.keys_map.get((keysym, mask), None)
         if key is None:
             logger.debug("Ignoring unknown keysym: %s, mask: %s", keysym, mask)
-            return
+            return (None, False)
```

Comment:
The bool here specifies whether the binding was handled, and if it wasn't then it couldn't be swallowed, right?

What I'm thinking is that the `tuple[Key | KeyChord | None, bool]` return type implies that `(None, True)` is a valid return value, but from what I can tell it wouldn't be valid. Could it be `tuple[Key | KeyChord, bool] | None` instead? This would also mean the change in `libqtile/backend/wayland/inputs.py` doesn't need the `[1]` index, making it a bit more clear when just reading the code in that file.

Reply:
But if we return ```(key, False)``` then before it would return ```False```, whereas with the change it would return ```True``` if I don't index by 1. This would only work for the ```None, False``` case where we can now just return ```None```.

 I could also make it ```Key | KeyChord | None``` but then I would have to set key to None conditionally on the ```key.swallow``` parameter which is weird when reading ```process_key_event```. What do you think?

Reply:
OK, nvm my suggestion actually. I think it's fine as is :)

Reply:
Great, thanks for thinking with me :+1: